stream.go

```diff
@@ -35,15 +35,61 @@ type Stream struct {
 	Conn *Connection
 }
 
-func fillBegin(enc *msgpack.Encoder, txnIsolation TxnIsolationLevel, timeout time.Duration) error {
-	hasTimeout := timeout > 0
-	hasIsolationLevel := txnIsolation != DefaultIsolationLevel
-	mapLen := 0
+// BeginRequest helps you to create a begin request object for execution
+// by a Stream.
+// Begin request can not be processed out of stream.
+type BeginRequest struct {
+	baseRequest
+	txnIsolation TxnIsolationLevel
+	timeout      time.Duration
+	isSync       *bool
```

Comment:
Let's avoud a pointer here.
```suggestion
	isSync        bool
	isSyncExist bool
```

Reply:
fixed